src/cdk/schematics/utils/build-component.ts

```diff
@@ -175,8 +175,13 @@ export function buildComponent(options: ComponentOptions,
     // Add the default component option values to the options if an option is not explicitly
     // specified but a default component option is available.
     Object.keys(options)
-      .filter(optionName => options[optionName] == null && defaultComponentOptions[optionName])
-      .forEach(optionName => options[optionName] = defaultComponentOptions[optionName]);
+      .filter((optionName: keyof ComponentOptions) => {
+        return options[optionName] == null &&
+               (defaultComponentOptions as ComponentOptions)[optionName];
```

Comment:
Have you considered updating `getDefaultComponentOptions` to return a `Partial<ComponentOptions>` or just `ComponentOptions`?